log/async_file_writer_test.go

```diff
@@ -1,24 +1,41 @@
 package log
 
 import (
-	"io/ioutil"
 	"os"
 	"strings"
 	"testing"
 )
 
-func TestWriter(t *testing.T) {
-	w := NewAsyncFileWriter("./hello.log", 100)
+func TestWriterHourly(t *testing.T) {
+	w := NewAsyncFileWriter("./hello.log", 100, false)
 	w.Start()
 	w.Write([]byte("hello\n"))
 	w.Write([]byte("world\n"))
 	w.Stop()
-	files, _ := ioutil.ReadDir("./")
+	files, _ := os.ReadDir("./")
 	for _, f := range files {
 		fn := f.Name()
 		if strings.HasPrefix(fn, "hello") {
 			t.Log(fn)
-			content, _ := ioutil.ReadFile(fn)
+			content, _ := os.ReadFile(fn)
+			t.Log(content)
+			os.Remove(fn)
+		}
+	}
+}
+
+func TestWriterDaily(t *testing.T) {
```

Comment:
I don't think these tests cases test if the ticker logic is correct - they only check that something was written to the log file. Can you think of a way to test the rotation logic? (I am thinking a more generic implementation with a variable rotation interval also has advantages in terms of ease of testing this logic).